packages/glimmer-runtime/tests/ember-component-test.ts

```diff
@@ -871,6 +871,59 @@ testComponent('parameterized has-block-params (concatted attr, default) when blo
   expected: '<button data-has-block-params="is-false"></button>'
 });
 
+module("Components - curlies - dynamic component");
+
+QUnit.test('missing definition', assert => {
```

Comment:
Can you also test it being initially false and becoming a component?  Also, the test implies that any lookup failures should be silent and just render blank.  Also, why is it rendering a comment?